api/src/mappers.js

```diff
@@ -0,0 +1,94 @@
+const PARAMETERS = {
+  '/items': ['code', 'exact', 'fields', 'fuzzy', 'inclusive', 'name', 'search'],
+};
+
+/**
+ * Parse query parameters.
+ *
+ * @param  {Object}        params    Request query parameters.
+ * @param  {Array<String>} whiteList Valid parameters per implementation.
+ * @return {Object}                  Object containing valid and invalid parameters.
+ */
+export const parseRequest = (request) => {
+  const { url } = request;
+  const whitelist = PARAMETERS[url];
+  const { params } = request;
+  return Object.keys(params).reduce(
+    ({ parameters }, param) => {
+      const { valid: validParameters, invalid: invalidParameters } = parameters;
+      if (whitelist.includes(param))
+        return { parameters: { valid: { ...validParameters, param }, invalid: invalidParameters } };
+      return { parameters: { valid: validParameters, invalid: { ...invalidParameters, param } } };
+    },
+    { parameters: { valid: [], invalid: [] } }
+  );
+};
+
+/**
+ * Map API request to GraphQL+- payload. Query parameters are assumed valid.
+ *
+ * @param  {Object} request REST api request.
+ * @return {String}         GraphQL+- response from Dgraph server.
```

Comment:
```suggestion
 * @return {String}         GraphQL+- request payload for Dgraph server.
```
Super nit-pick at the comment - at this point it's still a request payload, I think?

Reply:
Good point. 20940fe6a9416424f8d8ad774e84d7722dbbc614.